core/src/main/java/org/infinispan/transaction/tm/DummyTransaction.java

```diff
@@ -247,7 +250,7 @@ public void registerSynchronization(Synchronization sync) throws RollbackExcepti
    }
 
    public Collection<XAResource> getEnlistedResources() {
-      return Collections.unmodifiableList(resources);
+      return Collections.unmodifiableList(resources.stream().map(e -> e.getKey()).collect(Collectors.toList()));
```

Comment:
I see... but `getEnlistedResources()` is called all the time, e.g. in `runPrepare()`, so I'd rather not create a new list every time. I think `LinkedHashMap` might be a good compromise, if you need the resources to be kept in order. Or maybe we should just stop using `getEnlistedResources()` internally and have `endResources()` iterate on the list directly...